src/main/java/com/avioconsulting/log4j/sqs/SqsManager.java

```diff
@@ -80,112 +53,27 @@ public void startup() {
-                this.client = new AmazonSQSBufferedAsyncClient(asyncClient, config);
-                this.queueUrl = this.client.getQueueUrl(this.queueName).getQueueUrl();
-                if (largeMessagesEnabled) {
-                    logger.debug("Large Messages Enabled. Large Message Queue: " + this.largeMessageQueueName);
-                    this.largeMessageQueueUrl = this.client.getQueueUrl(this.largeMessageQueueName).getQueueUrl();
-                }
-            } catch (Exception e) {
-                logger.error("Failed to initialize SQS Client", e);
-                throw e;
-            }
+        if(messageLength <= maxMessageBytes ){
+            sendMessageList(ProcessorType.DEFAULT.name(),message,queueUrl,maxMessageBytes);
+        }else {
+            sendMessageList(largeMessageMode,message,queueUrl,maxMessageBytes);
         }
 
-        return this.client;
-    }
-
-    public String toString() {
-        return "[ region=" + this.awsRegion + ", maxBatchOpenMs=" + this.maxBatchOpenMs + ", maxBatchSize=" + this.maxBatchSize + ", maxInflightOutboundBatches=" + this.maxInflightOutboundBatches + " ]";
     }
 
-    /**
-     * Split the input string into a list of strings, each with a maximum memory size of maxsize bytes.
-     *
-     * @param src the source string
-     * @param encoding the encoding to use for the output string
-     * @param maxsize the maximum size of the string parts in bytes
-     * @return an array of strings, each with a maximum length of maxsize bytes
-     */
-    protected static String[] splitStringByByteLength(final String src, final String encoding, final int maxsize) {
-        Charset cs = Charset.forName(encoding);
-        CharsetEncoder coder = cs.newEncoder();
-        ByteBuffer out = ByteBuffer.allocate(maxsize);  // output buffer of required size
-        CharBuffer in = CharBuffer.wrap(src);
-        List<String> ss = new ArrayList<>();            // a list to store the chunks
-        int pos = 0;
-        while (true) {
-            CoderResult result = coder.encode(in, out, true); // try to encode as much as possible
-            int newpos = src.length() - in.length();
-            String s = src.substring(pos, newpos);
-            ss.add(s);                                  // add what has been encoded to the list
-            pos = newpos;                               // store new input position
-            out.rewind();                               // and rewind output buffer
-            if (!result.isOverflow()) {
-                break;                                  // everything has been encoded
-            }
-        }
-        return ss.toArray(new String[0]);
+    private void sendMessageList(String processorType, String message, String queueUrl, Integer maxMessageBytes) {
+        LogEventProcessor processor = ProcessorSupplier.selectProcessor(processorType);
```

Comment:
Would it be better to select the message processor at the beginning and then use that for everything? That way we're not loading the same class repeatedly.

Reply:
makes sense, let me check how I can improve that. Thanks for the suggestion.

Reply:
Did some improvements. However ProcessorSupplier has all the processors already instantiated ,so, depending on 'largeMessageMode' param,  supplier will return unique instance for each case. In scenario when message size < maxMessageBytes , Default processor is inyected, therefore, if it's needed, supplier has the capacity to retrieve other processor type depending on the situation (I added it to support further enhancements).